blog/posts/basics-navigation.md

```diff
@@ -37,10 +53,116 @@ node basic-browser-navigation.js
 :::
 ::::
 
-We also add `await browser.close()` to ensure that we are shutting down our browser before terminating the session. Without it, our execution would not return after the page has loaded, keeping our test hanging indefinitely.
+We also add `await browser.close()` to ensure that we are shutting down our browser before terminating the session. 
+Without it, our execution would not return after the page has loaded, keeping our test hanging indefinitely.
+
+## Waiting
+
+In your code, you have a range of options to wait for different things to happen in your browser session. There are a couple 
+that are very important:
+
+**page.waitForSelector()**
+
+This method waits for an element to appear in the page. This is your bread and butter and should be used whenever something 
+needs to be loaded after clicking, hovering, navigating etc. You can pass it an object with a `timeout` attribute 
+to override the default 30 seconds.
+
+In the example below, we type an email address into an input field on a login modal. Notice the difference between
+the Puppeteer and Playwright example.
+
+:::: tabs :options="{ useUrlFragment: false }"
+::: tab Puppeteer 
+<<< @/blog/snippets/puppeteer/basic-browser-waiting.js
+:::
+::: tab Playwright
+<<< @/blog/snippets/playwright/basic-browser-waiting.js
+:::
+::::
+
+Run this example as follows:
+:::: tabs :options="{ useUrlFragment: false }"
+::: tab MacOS
+```shell script
+node basic-browser-waiting.js
+```
+:::
+::: tab Windows
+```shell script
+node basic-browser-waiting.js
+```
+:::
+::::
+
+
+This works exactly the same for the `page.waitForXpath()` function is you are using Xpath selectors instead of CSS selectors.
+
+**page.waitForNavigation()**
+
+In your scripts you can click on a link that triggers a navigation to a new page. You can use Puppeteer's `page.waitForNavigation()` 
+method here to explicitly wait for this event to happen and then continue your script. The excepted notation in Puppeteer's
+case is by using the `Promise.all()` method to wait for the click to happen and the navigation to happen before continuing.
+
+```js
+const [response] = await Promise.all([
+  page.waitForNavigation(), 
+  page.click('a.some-link')
+]);
+```
+
+Interestingly, Playwright offers pretty much the same API for waiting on events and elements but again stresses its automatic handling 
+of the wait states under the hood. 
+
+::: tip
+Playwright handles a lot of the common waiting scenario's using its built in "auto waiting". Depending on your use case it
```

Comment:
Spelling: "built-in". Also, missing period at the end of the tip.

Reply:
Also, I would break up the last phrase like so:
"Depending on your use case, it might serve all your needs."